### PR TITLE
replace InstanceNotFoundException to allow building on Mono

### PR DESCRIPTION
**Pull request by devheroo**
It will be great to keep LuceneNet cross platform and compatible with Mono.
This is just a simple fix to make the build pass on Mono\Linux.

### CODE REVIEW

File: src/Lucene.Net.Facet/Taxonomy/WriterCache/CollisionMap.cs [hunk 1]
```diff
@@ -261,7 +261,7 @@ public Entry Next()
                 Entry e = this.next_Renamed;
                 if (e == null)
                 {
-                    throw new InstanceNotFoundException();
+					throw new Exception("InstanceNotFoundException");
```
Comment (synhershko) [inline src/Lucene.Net.Facet/Taxonomy/WriterCache/CollisionMap.cs]: Use `InvalidOperationException` or whatever Exception .NET IEnumerator throws on `Next` that fails

**Comment by synhershko** [comment]
Thanks! I've made a comment on the Exception type used

**Comment by devheroo** [comment]
Thanks for the tip , I made the change regarding the InvalidOperationException and also I had to rename a file path inside Lucene.Net.Codecs.Tests.csproj because files names are case sensitive in Linux and build wont pass without making this change.

**Comment by synhershko** [comment]
Thanks, merged. On your next PR, please work on a branch, it is an ill practice in git to merge from someone else's master.

**Comment by devheroo** [comment]
Thanks, that make sense, I will follow your suggestion next time :)

**Pull request merged**